bindata/network/multus-admission-controller/admission-controller.yaml

```diff
@@ -82,15 +82,29 @@ spec:
       - name: multus-admission-controller
         image: {{.MultusAdmissionControllerImage}}
         command:
-        - /usr/bin/webhook
-        args:
-        - -bind-address=0.0.0.0
-        - -port=6443
-        - -tls-private-key-file=/etc/webhook/tls.key
-        - -tls-cert-file=/etc/webhook/tls.crt
-        - -alsologtostderr=true
-        - -metrics-listen-address=127.0.0.1:9091
-        - -ignore-namespaces=openshift-etcd,openshift-console,openshift-ingress-canary,{{.IgnoredNamespace}}
+        - /bin/bash
+        - -c
+        - |-
+          set -euo pipefail
+{{- if .HyperShiftEnabled}}
+          retries=0
+          while [ ! -f /var/run/secrets/hosted_cluster/token ]; do
+            (( retries += 1 ))
+            sleep 1
+            if [[ "${retries}" -gt 30 ]]; then
+              echo "$(date -Iseconds) - Hosted cluster token not found"
+                exit 1
+            fi
+          done
+{{- end }}
```

Comment:
Wait.

/var/run/secrets/hosted_cluster/token is already used form the init container, why not wait there?

Also the wait fits better there I think since the init container is already expected to run to completion.

Reply:
The file is not read in the init container, the path is used to create a kubeconfig.
We cannot wait for it in the init container since it is actually created and managed by the sidecar container here: https://github.com/openshift/cluster-network-operator/blob/53b1e88d3e89886e94261a253748bf2898d2da6c/bindata/network/multus-admission-controller/admission-controller.yaml#L70